src/item.cpp

```diff
@@ -6432,7 +6502,7 @@ bool item::reload( player &u, item_location loc, int qty )
             qty = std::min( qty, ammo->ammo_remaining() );
             ammo->ammo_consume( qty, tripoint_zero );
             charges += qty;
-        } else if( ammo_type() == ammotype( "plutonium" ) ) {
+        } else if( ammo->type->ammo->type == "plutonium" ) {
```

Comment:
Looks like you are checking this same `ammo->type->ammo->type` thing quite often, and sometimes without checking to ensure `ammo->type->ammo` isn't a `cata::nullopt`. 

I'd recommend making a new function in item that checks if the item is ammo and then preforms this comparison.